nltk/twitter/util.py

```diff
@@ -0,0 +1,362 @@
+# -*- coding: utf-8 -*-
+# Natural Language Toolkit: Twitter client
+#
+# Copyright (C) 2001-2015 NLTK Project
+# Author: Ewan Klein <ewan@inf.ed.ac.uk>
+#         Lorenzo Rubio <lrnzcig@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+"""
+Utility functions to accompany :module:`twitterclient`.
+"""
+from __future__ import print_function
+import csv
+import json
+import os
+import pprint
+import nltk.compat as compat
+
+from twython import Twython
+
+
+def extract_fields(tweet, fields):
+    """
+    Extract field values from a full tweet and return them as a list
+
+    :param json tweet: The tweet in JSON format
+    :param list fields: The fields to be extracted from the tweet
+    :rtype: list(str)
+    """
+    out = []
+    for field in fields:
+        try:
+            _add_field_to_out(tweet, field, out)
+        except TypeError:
+            raise RuntimeError('Fatal error when extracting fields. Cannot find field ', field)
+    return out
+
+
+def _add_field_to_out(json, field, out):
+    if isinstance(field, dict):
+        for key, value in field.items():
+            _add_field_to_out(json[key], value, out)
+    else:
+        if isinstance(field, str):
+            out += [json[field]]
+        else:
+            out += [json[value] for value in field]
```

Comment:
In Python 2.x for unicode strings the second code path will be active. It is better to use `isinstance(field, compat.string_types)`.